### ISSUE TITLE
In-place computation of derivatives

### ISSUE THREAD

**Issue by ntezak** [JuliaLang/ODE.jl#95]
Hi there, I saw that there is already some discussion on memory efficiency [`here`](https://github.com/JuliaLang/ODE.jl/issues/69):
and a mention to allow for [` in-place RHS. `](https://github.com/JuliaLang/ODE.jl/pull/33#issuecomment-38195481)
This would be extremely valuable to me for the purpose of simulating quantum systems (and other high-dimensional problems such as PDEs) and so I was wondering how interested you are in making in-place ODEs the default.
I would be happy to work on this given your input.

**Comment by mauro3** [comment]
Yes, I agree, eg #72 uses in-place.  I think this can be done without disturbing the high level API.
